packages/gestalt/src/ComboBox.js

```diff
@@ -11,59 +11,132 @@ import {
   useMemo,
   useRef,
   useState,
+  type Ref,
 } from 'react';
 import Box from './Box.js';
 import Layer from './Layer.js';
 import Popover from './Popover.js';
 import Text from './Text.js';
 import InternalTextField from './InternalTextField.js';
 import Tag from './Tag.js';
-import ComboBoxItem, { type ComboBoxItemType } from './ComboBoxItem.js';
+import ComboBoxItem from './ComboBoxItem.js';
 import { ESCAPE, TAB, ENTER, UP_ARROW, DOWN_ARROW } from './keyCodes.js';
 import handleContainerScrolling, {
   KEYS,
   type DirectionOptionType,
 } from './utils/keyboardNavigation.js';
 
 type Size = 'md' | 'lg';
+type OptionType = {|
+  label: string,
+  subtext?: string,
+  value: string,
+|};
 
 type Props = {|
-  // REQUIRED
+  /**
+   * Label to describe the clear button's purpose.
+   */
   accessibilityClearButtonLabel: string,
+  /**
+   * Unique id to identify each ComboBox. Used for [accessibility](https://gestalt.pinterest.systems/combobox#Accessibility) purposes.
+   */
   id: string,
+  /**
+   * Provide a label to identify the ComboBox field.
+   */
   label: string,
-  options: $ReadOnlyArray<ComboBoxItemType>,
+  /**
+   * The data for each selection option. See [subtext](https://gestalt.pinterest.systems/combobox#With-subtext) variant to learn more.
+   */
+  options: $ReadOnlyArray<{|
+    label: string,
+    subtext?: string,
+    value: string,
+  |}>,
+  /**
+   * The text shown when the input value returns no matches.
+   */
   noResultText: string,
-  // OPTIONAL
+  /**
+   * When disabled, ComboBox looks inactive and cannot be interacted with. If tags are passed, they will appear disabled as well and cannot be removed. See [tags](https://gestalt.pinterest.systems/combobox#Tags) variant to learn more.
+   */
   disabled?: boolean,
```

Comment:
Default values aren't getting picked up. You'll need to move the props destructuring into the function signature for docgen to see the default values.

Reply:
Ah, makes sense. Thanks for adding them!